File: AAS.Domain/Bids/Bid.cs
using AAS.Domain.Bids.Enums;
using AAS.Tools.Types.IDs;

namespace AAS.Domain.Bids;

public class Bid
{
    public ID Id { get; }
    public Int32 Number { get; }
    public string Title { get; }
    public string? Description { get; }
    public string? DenyDescription { get; }
    public BidStatus Status { get; }
    public DateOnly? AcceptanceDate { get; }
    public DateOnly? ApproximateDate { get; }
    public ID CreatedUserId { get; }

    public Bid(ID id, Int32 number, string title, string? description, string? denyDescription, BidStatus status, DateOnly? acceptanceDate, DateOnly? approximateDate, ID createdUserId)
    {
        Id = id;
        Number = number;
        Title = title;
        Description = description;
        DenyDescription = denyDescription;
        Status = status;
        AcceptanceDate = acceptanceDate;
        ApproximateDate = approximateDate;
        CreatedUserId = createdUserId;
    }
}

File: AAS.BackOffice/client_app/src/domain/bids/bid.ts
export class Bid {
    constructor(
        public readonly id: string,
        public readonly number: number,
        public readonly title: string,
        public readonly description: string,
        public readonly denyDescription: string | null,
        public readonly status: number,
        public readonly acceptanceDate: Date | null,
        public readonly approxmateDate: Date | null
    ) { }
}

export function toBid(value: any): Bid {
    return new Bid(
        value.id,
        value.number,
        value.title,
        value.description,
        value.denyDescription,
        value.status,
        value.acceptanceDate,
        value.approxmateDate,
    )
}
export function toBids(values: any[]): Bid[] {
    return values.map(toBid);
}
